python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -79,18 +62,33 @@ def register_launch_commands(self):
                     )
                     continue
 
+            app_path = sw_entity[app_path_field]
             app_display_name = sw_entity["code"]
             app_args = sw_entity[app_args_field] or ""
-            register_cmd_data.extend(
-                self._build_register_command_data(
-                    app_display_name,
-                    local_thumb_path,
-                    app_engine,
-                    app_path,
-                    app_args,
-                    app_versions
-                )
+            app_icon_url = sw_entity["image"]
+
+            # Get the list of command data dictionaries from the information
+            # provided by this Software entity
+            command_data = self._build_register_command_data(
+                app_display_name, app_icon_url, app_engine, app_path, app_args, app_versions
```

Comment:
This doesn't seem clear - previously we passed in a path, now a url, only to compare it again once we have returned it back.

Could the shotgun thumbnail handling be moved into `_build_register_command_data` instead perhaps?